homeassistant/components/monoprice/__init__.py

```diff
@@ -57,11 +57,18 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
-    if unload_ok:
-        hass.data[DOMAIN][entry.entry_id][UNDO_UPDATE_LISTENER]()
+    if not unload_ok:
+        return False
+
+    hass.data[DOMAIN][entry.entry_id][UNDO_UPDATE_LISTENER]()
+
+    def _cleanup():
+        del hass.data[DOMAIN][entry.entry_id][MONOPRICE_OBJECT]
         hass.data[DOMAIN].pop(entry.entry_id)
 
-    return unload_ok
+    await hass.async_add_executor_job(_cleanup)
```

Comment:
Relying on thread safe operations of dict like this makes me nervous, how about something like this instead:

```py
    def _cleanup(monoprice: Monoprice) -> None:
        """Destroy the Monoprice object.

        Destroying the Monoprice closes the serial connection, do it in an executor so the garbage
        collection does not block.
        """
        del monoprice

    monoprice: Monoprice = hass.data[DOMAIN][entry.entry_id][MONOPRICE_OBJECT]
    hass.data[DOMAIN].pop(entry.entry_id)
    await hass.async_add_executor_job(_cleanup)
```

Reply:
Thanks. Changed according to your comments, but left the type hint for monoprice out for now because we don't use it anywhere in the integration. I have plans to fork the library and refactor it, after which I'll end type hints (and this piece of code won't be needed)